src/PIL/ImageTk.py

```diff
@@ -142,6 +131,8 @@ def __init__(
             self.paste(image)
 
     def __del__(self) -> None:
+        if not hasattr(self, "__photo"):
+            return
```

Comment:
I don't think this works as simply as one might imagine.

```python
class A:
	def __init__(self):
		self.photo = True
		self._photo = True
		self.__photo = True
		
		print([
			hasattr(self, "photo"), # True
			hasattr(self, "_photo"), # True
			hasattr(self, "__photo") # False
		])

A()
```

But I'm also curious - what made you think there would be a point in the lifecycle when the object didn't have `__photo`?

Reply:
> what made you think there would be a point in the lifecycle when the object didn't have __photo

This happens when `ImageTk.PhotoImage()` returns `ValueError`.

```
$ pytest Tests/test_imagetk.py
===================================================== test session starts ======================================================
platform darwin -- Python 3.9.13, pytest-8.3.2, pluggy-1.5.0
--------------------------------------------------------------------
Pillow 11.0.0.dev0
Python 3.9.13 (v3.9.13:6de2ca5339, May 17 2022, 11:37:23)
       [Clang 13.0.0 (clang-1300.0.29.30)]
--------------------------------------------------------------------
Python executable is /Users/Master/Code/_env/env_py39/bin/python3
Environment Python files loaded from /Users/Master/Code/_env/env_py39
System Python files loaded from /Library/Frameworks/Python.framework/Versions/3.9
--------------------------------------------------------------------
Python Pillow modules loaded from /Users/Master/Code/Pillow/src/PIL
Binary Pillow modules loaded from /Users/Master/Code/Pillow/src/PIL
--------------------------------------------------------------------
--- PIL CORE support ok, compiled for 11.0.0.dev0
--- TKINTER support ok, loaded 8.6
*** FREETYPE2 support not installed
--- LITTLECMS2 support ok, loaded 2.16
--- WEBP support ok, loaded 1.4.0
--- JPEG support ok, compiled for libjpeg-turbo 3.0.3
*** OPENJPEG (JPEG2000) support not installed
--- ZLIB (PNG/ZIP) support ok, loaded 1.2.11
--- LIBTIFF support ok, loaded 4.6.0
*** RAQM (Bidirectional Text) support not installed
*** LIBIMAGEQUANT (Quantization method) support not installed
*** XCB (X protocol) support not installed
--------------------------------------------------------------------

rootdir: /Users/Master/Code/Pillow
configfile: pyproject.toml
plugins: cov-5.0.0, env-0.6.2, timeout-2.3.1, django-4.5.2, xdist-3.6.1
collected 13 items                                                                                                             

Tests/test_imagetk.py .............                                                                                      [100%]

======================================================= warnings summary =======================================================
Tests/test_imagetk.py::test_photoimage[1]
Tests/test_imagetk.py::test_photoimage[L]
Tests/test_imagetk.py::test_photoimage[P]
Tests/test_imagetk.py::test_photoimage[RGB]
Tests/test_imagetk.py::test_photoimage[RGBA]
  /Users/Master/Code/_env/env_py39/lib/python3.9/site-packages/_pytest/unraisableexception.py:85: PytestUnraisableExceptionWarning: Exception ignored in: <function PhotoImage.__del__ at 0x102963430>
  
  Traceback (most recent call last):
    File "/Users/Master/Code/Pillow/src/PIL/ImageTk.py", line 134, in __del__
      name = self.__photo.name
  AttributeError: 'PhotoImage' object has no attribute '_PhotoImage__photo'
  
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================ 13 passed, 5 warnings in 0.24s ================================================
```

Reply:
I've fixed `hasattr` here